garcon/decider.py

```diff
@@ -196,7 +197,20 @@ def run(self):
                 running process.
         """
 
-        poll = self.poll()
+        try:
+            poll = self.poll()
+        except Exception as error:
+            # Catch exceptions raised during poll() to avoid a Decider thread
+            # dying & the daemon unable to process subsequent workflows.
+            # on_exception() can be overriden by the flow to send an alert
+            # when such an exception occurs.
+            if self.on_exception:
+                self.on_exception(self, error)
+            else:
+                traceback.print_stack()
+                print('Exception: {}'.format(error))
```

Comment:
Wasn't sure of the best way to deal with this since the Decider doesn't have a logger. To help a user debug it seems like we still have to at least print out the exception even if a custom `on_exception()` method hasn't been attached to the workflow. Definitely open to alternatives.